plugin/forward/connect.go

```diff
@@ -44,29 +44,29 @@ func (t *Transport) updateDialTimeout(newDialTime time.Duration) {
 }
 
 // Dial dials the address configured in transport, potentially reusing a connection or creating a new one.
-func (t *Transport) Dial(proto string) (*dns.Conn, bool, error) {
+func (t *Transport) Dial(proto string) (*persistConn, bool, error) {
 	// If tls has been configured; use it.
 	if t.tlsConfig != nil {
 		proto = "tcp-tls"
 	}
 
 	t.dial <- proto
-	c := <-t.ret
+	pc := <-t.ret
 
-	if c != nil {
-		return c, true, nil
+	if pc != nil {
+		return pc, true, nil
 	}
 
 	reqTime := time.Now()
 	timeout := t.dialTimeout()
 	if proto == "tcp-tls" {
 		conn, err := dns.DialTimeoutWithTLS("tcp", t.addr, t.tlsConfig, timeout)
 		t.updateDialTimeout(time.Since(reqTime))
-		return conn, false, err
+		return &persistConn{conn, time.Now()}, false, err
```

Comment:
We should update `used` field in `func (t *Transport) Yield(pc *persistConn)` before sending to channel. This update should take place for all connections, not only to newly created.

Reply:
better is to rename used to created - not sure what my original intention were, but a def. lifetime was some part of it. I'll rename the field to reflect this - as this PR actually implement that.

Reply:
`used` is the time when this connection was last used. We need this time to detect expired connections, i.e. connection is considered expired when some time elapsed after last usage of this connection. And this is the reason why we need to update this field after each use of it.

Reply:
done - pushing in a minute